contracts/cryptography/ECDSA.sol

```diff
@@ -33,9 +33,13 @@ library ECDSA {
             v := byte(0, mload(add(signature, 0x60)))
         }
 
-        // Version of signature should be 27 or 28, but 0 and 1 are also possible versions
-        if (v < 27) {
-            v += 27;
+        //EIP-2 still allows signature malleabality for ecrecover(), remove this possibility and make
+        //the signature unique. Most signatures from current libraries are generate a unique signature with
+        //an s-value in the lower half order. If not, calculate a new s-value with
+        //0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141 - s1 and flip v from 27 to 28 or
+        //vice versa. It could also happen that a signature uses 0/1 instead 27/28 for v. In that case add 27.
+        if(uint256(s) > uint256(0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0)) {
```

Comment:
Is there a friendlier way of writing this number?

Reply:
Also, could we add a link to Appendix F of the Ethereum Yellow paper somewhere in the comments?

Reply:
I added the link to the yellow paper and changed the comment. With respect to the friendlier way: others and e.g., [geth](https://github.com/ethereum/go-ethereum/blob/master/crypto/crypto.go) also defines it as:
```
secp256k1N, _  = new(big.Int).SetString("fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141", 16)
secp256k1halfN = new(big.Int).Div(secp256k1N, big.NewInt(2))
```
I think we should leave it as is.